### PR TITLE
python3Packages.accelerate: Bump version 1.0.0 -> 1.1.0 and fix few things

### PR DESCRIPTION
**Pull request by mikatammi**
python3Packages.accelerate
* Add missing huggingface-hub dependency, it caused package being broken already before version bump
* Bump version 1.0.0 -> 1.1.0
* On Darwin: Disable some tests, which started failing after enabling MPS in pytorch package

Changelog can be found here: https://github.com/huggingface/accelerate/releases

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [X] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)

Tested some of the basic functionality.

I tested that now running binaries don't give straight ImportError about missing huggingface-hub anymore, but didn't test the functionality otherwise than trying to run binaries does print out --help to the terminal. I'm able to use accelerate in my own Python code whch runs StableDiffusion.

- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by mikatammi** [comment]
Here's log of accelerate tests failing:

[accelerate-test-fails.txt](https://github.com/user-attachments/files/17607854/accelerate-test-fails.txt)

File: pkgs/development/python-modules/accelerate/default.nix [hunk 1]
```diff
@@ -113,6 +115,31 @@ buildPythonPackage rec {
       "test_init_trackers"
       "test_log"
       "test_log_with_tensor"
+
+      # After enabling MPS in pytorch, these tests started failing
```
Comment (dotlambda) [inline pkgs/development/python-modules/accelerate/default.nix:119]: If they only fail on Darwin, they should only be disabled there.

Comment (mikatammi) [reply pkgs/development/python-modules/accelerate/default.nix:119]: Yes, that's why they are in the block which checks for stdenv.hostPlatform.isDarwin

**Comment by mikatammi** [comment]
Two questions i'm mostly concerned about:
* After doing version bump, should we check if some tests started working again?
* Should we report about some of the failing tests to accelerate upstream?

File: pkgs/development/python-modules/accelerate/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/accelerate/default.nix:119]: using `++ lib.optionals stdenv.hostPlatform.isDarwin [ ... ];`.

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353184`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 4 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 18 packages failed to build:</summary>
  <ul>
    <li>openllm</li>
    <li>openllm.dist</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python311Packages.cltk</li>
    <li>python311Packages.cltk.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.llama-index-embeddings-huggingface</li>
    <li>python311Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python311Packages.txtai</li>
    <li>python311Packages.txtai.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 46 packages built:</summary>
  <ul>
    <li>libretranslate (python312Packages.libretranslate)</li>
    <li>libretranslate.dist (python312Packages.libretranslate.dist)</li>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>python311Packages.accelerate</li>
    <li>python311Packages.accelerate.dist</li>
    <li>python311Packages.argos-translate-files</li>
    <li>python311Packages.argos-translate-files.dist</li>
    <li>python311Packages.argostranslate</li>
    <li>python311Packages.argostranslate.dist</li>
    <li>python311Packages.langchain-huggingface</li>
    <li>python311Packages.langchain-huggingface.dist</li>
    <li>python311Packages.libretranslate</li>
    <li>python311Packages.libretranslate.dist</li>
    <li>python311Packages.manifest-ml</li>
    <li>python311Packages.manifest-ml.dist</li>
    <li>python311Packages.peft</li>
    <li>python311Packages.peft.dist</li>
    <li>python311Packages.sentence-transformers</li>
    <li>python311Packages.sentence-transformers.dist</li>
    <li>python311Packages.stanza</li>
    <li>python311Packages.stanza.dist</li>
    <li>python311Packages.translatehtml</li>
    <li>python311Packages.translatehtml.dist</li>
    <li>python312Packages.accelerate</li>
    <li>python312Packages.accelerate.dist</li>
    <li>python312Packages.argos-translate-files</li>
    <li>python312Packages.argos-translate-files.dist</li>
    <li>python312Packages.argostranslate</li>
    <li>python312Packages.argostranslate.dist</li>
    <li>python312Packages.k-diffusion</li>
    <li>python312Packages.k-diffusion.dist</li>
    <li>python312Packages.langchain-huggingface</li>
    <li>python312Packages.langchain-huggingface.dist</li>
    <li>python312Packages.llama-index-embeddings-huggingface</li>
    <li>python312Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python312Packages.manifest-ml</li>
    <li>python312Packages.manifest-ml.dist</li>
    <li>python312Packages.peft</li>
    <li>python312Packages.peft.dist</li>
    <li>python312Packages.sentence-transformers</li>
    <li>python312Packages.sentence-transformers.dist</li>
    <li>python312Packages.stanza</li>
    <li>python312Packages.stanza.dist</li>
    <li>python312Packages.translatehtml</li>
    <li>python312Packages.translatehtml.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 12 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>openllm</li>
    <li>openllm.dist</li>
    <li>python311Packages.txtai</li>
    <li>python311Packages.txtai.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 50 packages built:</summary>
  <ul>
    <li>libretranslate (python312Packages.libretranslate)</li>
    <li>libretranslate.dist (python312Packages.libretranslate.dist)</li>
    <li>python311Packages.accelerate</li>
    <li>python311Packages.accelerate.dist</li>
    <li>python311Packages.argos-translate-files</li>
    <li>python311Packages.argos-translate-files.dist</li>
    <li>python311Packages.argostranslate</li>
    <li>python311Packages.argostranslate.dist</li>
    <li>python311Packages.cltk</li>
    <li>python311Packages.cltk.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.langchain-huggingface</li>
    <li>python311Packages.langchain-huggingface.dist</li>
    <li>python311Packages.libretranslate</li>
    <li>python311Packages.libretranslate.dist</li>
    <li>python311Packages.llama-index-embeddings-huggingface</li>
    <li>python311Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python311Packages.manifest-ml</li>
    <li>python311Packages.manifest-ml.dist</li>
    <li>python311Packages.peft</li>
    <li>python311Packages.peft.dist</li>
    <li>python311Packages.sentence-transformers</li>
    <li>python311Packages.sentence-transformers.dist</li>
    <li>python311Packages.stanza</li>
    <li>python311Packages.stanza.dist</li>
    <li>python311Packages.translatehtml</li>
    <li>python311Packages.translatehtml.dist</li>
    <li>python312Packages.accelerate</li>
    <li>python312Packages.accelerate.dist</li>
    <li>python312Packages.argos-translate-files</li>
    <li>python312Packages.argos-translate-files.dist</li>
    <li>python312Packages.argostranslate</li>
    <li>python312Packages.argostranslate.dist</li>
    <li>python312Packages.k-diffusion</li>
    <li>python312Packages.k-diffusion.dist</li>
    <li>python312Packages.langchain-huggingface</li>
    <li>python312Packages.langchain-huggingface.dist</li>
    <li>python312Packages.llama-index-embeddings-huggingface</li>
    <li>python312Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python312Packages.manifest-ml</li>
    <li>python312Packages.manifest-ml.dist</li>
    <li>python312Packages.peft</li>
    <li>python312Packages.peft.dist</li>
    <li>python312Packages.sentence-transformers</li>
    <li>python312Packages.sentence-transformers.dist</li>
    <li>python312Packages.stanza</li>
    <li>python312Packages.stanza.dist</li>
    <li>python312Packages.translatehtml</li>
    <li>python312Packages.translatehtml.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:fast_forward: 50 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>libretranslate</li>
    <li>libretranslate.dist</li>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python311Packages.argos-translate-files</li>
    <li>python311Packages.argos-translate-files.dist</li>
    <li>python311Packages.argostranslate</li>
    <li>python311Packages.argostranslate.dist</li>
    <li>python311Packages.langchain-huggingface</li>
    <li>python311Packages.langchain-huggingface.dist</li>
    <li>python311Packages.libretranslate</li>
    <li>python311Packages.libretranslate.dist</li>
    <li>python311Packages.llama-index-embeddings-huggingface</li>
    <li>python311Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python311Packages.manifest-ml</li>
    <li>python311Packages.manifest-ml.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python311Packages.sentence-transformers</li>
    <li>python311Packages.sentence-transformers.dist</li>
    <li>python311Packages.translatehtml</li>
    <li>python311Packages.translatehtml.dist</li>
    <li>python311Packages.txtai</li>
    <li>python311Packages.txtai.dist</li>
    <li>python312Packages.argos-translate-files</li>
    <li>python312Packages.argos-translate-files.dist</li>
    <li>python312Packages.argostranslate</li>
    <li>python312Packages.argostranslate.dist</li>
    <li>python312Packages.k-diffusion</li>
    <li>python312Packages.k-diffusion.dist</li>
    <li>python312Packages.langchain-huggingface</li>
    <li>python312Packages.langchain-huggingface.dist</li>
    <li>python312Packages.libretranslate</li>
    <li>python312Packages.libretranslate.dist</li>
    <li>python312Packages.llama-index-embeddings-huggingface</li>
    <li>python312Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python312Packages.manifest-ml</li>
    <li>python312Packages.manifest-ml.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python312Packages.sentence-transformers</li>
    <li>python312Packages.sentence-transformers.dist</li>
    <li>python312Packages.translatehtml</li>
    <li>python312Packages.translatehtml.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>openllm</li>
    <li>openllm.dist</li>
    <li>python311Packages.cltk</li>
    <li>python311Packages.cltk.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 12 packages built:</summary>
  <ul>
    <li>python311Packages.accelerate</li>
    <li>python311Packages.accelerate.dist</li>
    <li>python311Packages.peft</li>
    <li>python311Packages.peft.dist</li>
    <li>python311Packages.stanza</li>
    <li>python311Packages.stanza.dist</li>
    <li>python312Packages.accelerate</li>
    <li>python312Packages.accelerate.dist</li>
    <li>python312Packages.peft</li>
    <li>python312Packages.peft.dist</li>
    <li>python312Packages.stanza</li>
    <li>python312Packages.stanza.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:fast_forward: 28 packages marked as broken and skipped:</summary>
  <ul>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>libretranslate</li>
    <li>libretranslate.dist</li>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python311Packages.cltk</li>
    <li>python311Packages.cltk.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.libretranslate</li>
    <li>python311Packages.libretranslate.dist</li>
    <li>python311Packages.llama-index-embeddings-huggingface</li>
    <li>python311Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python311Packages.private-gpt</li>
    <li>python311Packages.private-gpt.dist</li>
    <li>python312Packages.k-diffusion</li>
    <li>python312Packages.k-diffusion.dist</li>
    <li>python312Packages.libretranslate</li>
    <li>python312Packages.libretranslate.dist</li>
    <li>python312Packages.llama-index-embeddings-huggingface</li>
    <li>python312Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>openllm</li>
    <li>openllm.dist</li>
    <li>python311Packages.txtai</li>
    <li>python311Packages.txtai.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 36 packages built:</summary>
  <ul>
    <li>python311Packages.accelerate</li>
    <li>python311Packages.accelerate.dist</li>
    <li>python311Packages.argos-translate-files</li>
    <li>python311Packages.argos-translate-files.dist</li>
    <li>python311Packages.argostranslate</li>
    <li>python311Packages.argostranslate.dist</li>
    <li>python311Packages.langchain-huggingface</li>
    <li>python311Packages.langchain-huggingface.dist</li>
    <li>python311Packages.manifest-ml</li>
    <li>python311Packages.manifest-ml.dist</li>
    <li>python311Packages.peft</li>
    <li>python311Packages.peft.dist</li>
    <li>python311Packages.sentence-transformers</li>
    <li>python311Packages.sentence-transformers.dist</li>
    <li>python311Packages.stanza</li>
    <li>python311Packages.stanza.dist</li>
    <li>python311Packages.translatehtml</li>
    <li>python311Packages.translatehtml.dist</li>
    <li>python312Packages.accelerate</li>
    <li>python312Packages.accelerate.dist</li>
    <li>python312Packages.argos-translate-files</li>
    <li>python312Packages.argos-translate-files.dist</li>
    <li>python312Packages.argostranslate</li>
    <li>python312Packages.argostranslate.dist</li>
    <li>python312Packages.langchain-huggingface</li>
    <li>python312Packages.langchain-huggingface.dist</li>
    <li>python312Packages.manifest-ml</li>
    <li>python312Packages.manifest-ml.dist</li>
    <li>python312Packages.peft</li>
    <li>python312Packages.peft.dist</li>
    <li>python312Packages.sentence-transformers</li>
    <li>python312Packages.sentence-transformers.dist</li>
    <li>python312Packages.stanza</li>
    <li>python312Packages.stanza.dist</li>
    <li>python312Packages.translatehtml</li>
    <li>python312Packages.translatehtml.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
> * After doing version bump, should we check if some tests started working again?

Ideally yes. You can try re-enabling them all and adding them back only if needed.



> Should we report about some of the failing tests to accelerate upstream?

Also yes, if the failures are not due to our testing setup (example, no need to report tests that are failing because of lacking internet access).

**Review by GaetanLepage — APPROVED** [review]
LGTM

**Pull request merged**